src/main/java/homework/lectures/lecture7_collections/shop/Shop.java

```diff
@@ -0,0 +1,62 @@
+package homework.lectures.lecture7_collections.shop;
+
+import org.apache.log4j.Logger;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Shop {
+    private static final Logger logger = Logger.getLogger(Shop.class);
+    private final Map<String, BigDecimal> shopMap = new HashMap<>();
+    private Product product;
+
+    public Shop() {
+    }
+
+    public Shop(Product product) {
+        this.product = product;
+        shopMap.put(product.getProductName(), product.getProductCost());
+    }
+
+    public void addProduct(Product product) {
+        shopMap.put(product.getProductName(), product.getProductCost());
+        logger.info("The product " + product.getProductName() + " was added!");
+    }
+
+    public void deleteProduct(Product product) {
+        shopMap.remove(product.getProductName());
+        logger.info("The product " + product.getProductName() + " was deleted");
+    }
+
+    public void findProductByName(String productName) {
+        boolean isFound = false;
```

Comment:
`findProductByName` that OK but why this method return nothing(void)
I expected that this method return Book with name which I get in brackets like parameter

Reply:
And try did it without flag true or false

Reply:
done